tasks/services/works.js

```diff
@@ -89,6 +94,112 @@ function createOrderPlans (hook) {
     .then(() => hook)
 }
 
+function sendOrderStartEmail (credential, appConfig, mailer, order) {
+  const orderName = order.name ? order.name : `Order ${order.id}`
+  return mailer.create({
+    from: `${appConfig.email}`,
+    to: credential.email || 'no@email.com',
+    subject: `An order has been started on ${appConfig.name}!`,
+    html: `
+      Hi. You're invited to join an order, ${orderName}, on ${appConfig.name}!
+
+      <br />
+      <br />
+
+      ${appConfig.bodyText}
+
+      <br />
+      <br />
+
+      Click <a href=${appConfig.url}/o/${order.id}>here</a> to join the order!
+    `
+  })
+}
+
+function sendWelcomeEmail (credential, appConfig, mailer, order, token) {
+  return mailer.create({
+    from: `${appConfig.email}`,
+    to: credential.email || 'no@email.com',
+    subject: `You're invited to join ${appConfig.name}!`,
+    html: `
+      Hi. You're invited to join a group on ${appConfig.name}!
+
+      <br />
+      <br />
+
+      ${appConfig.bodyText}
+
+      <br />
+      <br />
+
+      Click <a href=${appConfig.url}/invited/${token.jwt}>here</a> to set your password and start buying together!
+    `
+  })
+}
+
+function sendEmailBasedOnPasswordStatus (hook, order) {
+  const tokens = hook.app.service('tokens')
+  const appConfig = hook.app.get('app')
+  const mailer = hook.app.service('mailer')
+
+  const hasPassword = (credential) => {
+    return not(isNil(credential.password))
+  }
+
+  return (credential) => {
+    if (hasPassword(credential)) {
+      return sendOrderStartEmail(credential, appConfig, mailer, order)
+    } else {
+      return tokens.create({
+        agentId: credential.agentId,
+        service: 'credentials',
+        method: 'patch',
+        params: { serviceId: credential.id }
+      })
+      .then((token) => {
+        return sendWelcomeEmail(credential, appConfig, mailer, order, token)
+      })
+    }
+  }
+}
+
+function sendStartOrderEmails (hook) {
+  const taskPlans = hook.app.service('taskPlans')
+  const orders = hook.app.service('orders')
+  const relationships = hook.app.service('relationships')
+  const credentials = hook.app.service('credentials')
+
+  return taskPlans.get(hook.data.taskPlanId)
+  .then((taskPlanResult) => {
+    return orders.get(taskPlanResult.params.orderId)
+    .then((orderResult) => {
```

Comment:
can this be part of parent promise chain? doesn't seem like we need `taskPlanResult` again